docs/dev-info/style-guide.md

```diff
@@ -0,0 +1,75 @@
+# Guppy Style Guide
+
+We rely on tools like Flow, Prettier, and ESLint to handle most "code style" concerns. However, even with these tools, some things still need to be done manually.
+
+This document holds our conventions around code styles.
+
+---
+
+### Imports
+
+Imports should be grouped in the following way:
+
+```js
+// First block: Native Node modules and third-party modules
+import fs from 'fs';
+import React, { Component } from 'react';
+
+// Second block: actions, constants, reducers, services, etc
+import { refreshProjects } from '../../actions';
+import { COLORS } from '../../constants';
+import { getOnboardingStatus } from '../../reducers/onboarding-status.reducer';
+
+// Third block: React components
+// NOTE: Second and third block can be combined in files without a ton of
+// imports.
+import Sidebar from '../Sidebar';
+import ProjectPage from '../ProjectPage';
+
+// Final block: type imports, both third-party and Guppy-specific
+import type {Action} from 'redux';
+import type {Project} from '../../types';
+```
+
+Having consistent import groups makes it easy to quickly get a sense of which
+modules a file is using.
+
+### React component structure
+
+Components should have their props and state defined, when needed, as specific variables:
+
+```js
+// BAD
+class BadExample extends Component<{ children: React$Node }> {
+  ...
+}
+
+// GOOD
+type Props = {
```

Comment:
Are there ever multiple components in a file? If so, it might be better to use something like `GoodExampleProps` instead of plain `Props`.

Reply:
Good call-out! I'm actually not a fan of having multiple stateful components in a file, and for SFCs I'm fine if the props are inlined. I'll add all this info to the doc.